src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -51,32 +51,51 @@ public boolean extractQuestionDetails(
         
         int numOfRubricChoices = Integer.parseInt(numOfRubricChoicesString);
         int numOfRubricSubQuestions = Integer.parseInt(numOfRubricSubQuestionsString);
-        List<String> rubricChoices = new ArrayList<String>();
-        List<String> rubricSubQuestions = new ArrayList<String>();
-        List<List<String>> rubricDescriptions = new ArrayList<List<String>>();
+        List<String> rubricChoices = getRubricChoices(requestParameters, numOfRubricChoices);
+        List<String> rubricSubQuestions = getSubQuestions(requestParameters, numOfRubricSubQuestions);
+        List<List<String>> rubricDescriptions = getRubricQuestionDescriptions(requestParameters, 
+                                                                              numOfRubricChoices, numOfRubricSubQuestions);
         
-        int numActualChoices = 0;
-        int numActualSubQuestions = 0;
+        int numActualChoices = rubricChoices.size();
+        int numActualSubQuestions = rubricSubQuestions.size();
         
-        // Get list of choices
+        // Set details
+        setRubricQuestionDetails(numActualChoices, rubricChoices, numActualSubQuestions, 
+                                 rubricSubQuestions, rubricDescriptions);
```

Comment:
I think `numActualChoices` and `numActualSubQuestions` are redundant here. On `setRubricQuestionDetails` you can get the sizes of `rubricChoices` and `rubricSubQuestion` as needed.